Apps/WebClient/src/Server/Controllers/UserProfileController.cs

```diff
@@ -320,5 +320,48 @@ public async Task<IActionResult> UpdateUserSMSNumber(string hdid, [FromBody] str
             bool result = this.userSMSService.UpdateUserSMS(hdid, smsNumber, new Uri(referer), bearerToken);
             return new JsonResult(result);
         }
+
+        /// <summary>
+        /// Gets the user preference.
+        /// </summary>
+        /// <returns>The user preference model wrapped in a request result.</returns>
+        /// <param name="hdid">The user hdid.</param>
+        /// <response code="200">Returns the user preference json.</response>
+        /// <response code="401">the client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpGet]
+        [Route("{hdid}/userpreference")]
+        [Authorize(Policy = UserPolicy.Read)]
+        public IActionResult GetUserPreference(string hdid)
```

Comment:
Although it could be a stand alone resource, it looks to me that every time we need the user preference we will use the user profile

Reply:
right, for me, user preference should be a part of user profile. also, I was thinking to have them all in 1 call between client and server.
Are you suggesting we should separate the user preference from user profile now?

Reply:
nmv, I have just seen your "lets move the preferences to the same object. That way a single call will load both" above then.